src/http/api/views/reply.chain.view.ts

```diff
@@ -0,0 +1,357 @@
+            title: result.post_title ?? '',
+            excerpt: result.post_excerpt ?? '',
+            summary: result.post_summary ?? null,
+            content: result.post_content ?? '',
+            url: result.post_url,
+            featureImageUrl: result.post_image_url ?? null,
+            publishedAt: result.post_published_at,
+            likeCount: result.post_like_count,
+            likedByMe: result.post_liked_by_current_user === 1,
+            replyCount: result.post_reply_count,
+            readingTimeMinutes: result.post_reading_time_minutes,
+            attachments: result.post_attachments
+                ? result.post_attachments.map((attachment) => ({
+                      type: attachment.type ?? '',
+                      mediaType: attachment.mediaType ?? '',
+                      name: attachment.name ?? '',
+                      url: attachment.url,
+                  }))
+                : [],
+            author: {
+                id: result.author_id.toString(),
+                handle: getAccountHandle(
+                    result.author_url ? new URL(result.author_url).host : '',
+                    result.author_username,
+                ),
+                name: result.author_name ?? '',
+                url: result.author_url ?? '',
+                avatarUrl: result.author_avatar_url ?? '',
+            },
+            authoredByMe: result.author_id === contextAccountId,
+            repostCount: result.post_repost_count,
+            repostedByMe: result.post_reposted_by_current_user === 1,
+            repostedBy: null,
+        };
+    }
+
+    private async getAncestors(
+        contextAccountId: number,
+        postApId: URL,
+    ): Promise<PostRow[]> {
+        const db = this.db;
+        const selectPostRow = this.selectPostRow(contextAccountId);
+        const ancestorRows = await selectPostRow(
+            db
+                .withRecursive('ancestor_ids', (qb) => {
+                    qb.select('id', 'in_reply_to', db.raw('0 AS depth'))
+                        .from('posts')
+                        .where('ap_id', postApId.href)
+                        .unionAll(function () {
+                            this.select(
+                                'p.id',
+                                'p.in_reply_to',
+                                db.raw('ai.depth + 1'),
+                            )
+                                .from('ancestor_ids as ai')
+                                .join('posts as p', 'p.id', 'ai.in_reply_to')
+                                .whereNotNull('ai.in_reply_to')
+                                .andWhere(
+                                    'ai.depth',
+                                    '<',
+                                    ReplyChainView.MAX_ANCESTOR_DEPTH,
+                                );
+                        });
+                })
+                .from('ancestor_ids')
+                .where('depth', '>', 0)
```

Comment:
This means we don't both getting the current post, which is at depth 0 - only the ancestors